Add support for running Appium tests with BrowserStack

BretJohnson opened the pull request:
This PR also bumps Appium.WebDriver from 5.0.0-beta02
to 5.0.0-rc.2, both for local runs and BrowserStack runs.

Commit:
Add test projects, with a separate assembly for Android and iOS

Commit:
Bump Appium WebDriver to latest (rc1 build)

Commit:
Downgrade to net7.0 as BS currently has an issue with net8.0

The BrowserStack TestAdapter SDK currently doesn't support
net8.0, since it uses MonoMod which currently has this
issue on net8.0: https://github.com/MonoMod/MonoMod/issues/136.

Commit:
Update browserstack.yml

Commit:
Add USE_BROWSERSTACK define

Currently the BrowserStack TestAdapter SDK doesn't support .NET8.
It's blocked by this issue: https://github.com/MonoMod/MonoMod/issues/136

So for now we build it with .NET7 and use #if !USE_BROWSERSTACK to remove
any references to code in MAUI itself from the Appium tests (which a handful
of tests require).

In practice this means like like 99% of tests will run on BrowserStack
initially, which is likely good enough. Once the .net8 issue is resolved
it can be 100%.

jsuarezruiz reviewed (commented):
Nice, could you also create an associated task to update https://github.com/dotnet/maui/wiki/UITests?

Commit:
Recreate Appium connection for each test

For BrowserStack, the Appium driver should get
recreated for each test, so that each tests is in
a separate BrowserStack session.

Commit:
Update settings in browserstack.yml

Commit:
Restore [Test] methods in base classes as they work now

Commit:
Bump BrowserStack SDK to the latest, 0.4.1

Commit:
Include build changes to have a deployable debug APK

Commit:
Add RunTestsInIsolationProperty and rework some code accordingly

Commit:
Use Quit instead of Dispose, temporarily, to make work with BStack

jfversluis committed:
Rename iOS folder to match Android

jfversluis committed:
Get iOS bits ready

Commit:
Add missing RunTestsInIsolation check

Commit:
Bump BStack SDK

Bump to latest BrowserStack SDK version (as of a week ago when I made this update).
Also remove the workaround disabling treating warnings as errors, as that's no longer needed with the latest BStack SDK.

Commit:
Update handling around RunTestsInIsolation property

Commit:
Switch back to 5.0.0-beta02 Appium driver

Commit:
Bump to latest BrowserStack.TestAdapter

Commit:
Bump to latest BStack TestAdapter and net8, now supported

Commit:
Bump to 5.0.0-rc.2 Appium driver (needed for BStack test status to work properly)

Commit:
Merge branch 'main' into use-browserstack-for-appium-tests

Commit:
Merge branch 'main' into use-browserstack-for-appium-tests

Commit:
Share build targets for appium projects

Also add BrowserStack Android test project, in new local with shared project settings.

Commit:
Merge branch 'main' into use-browserstack-for-appium-tests

Commit:
Add AppiumTests shared code

Commit:
Bump BrowserStack version

Commit:
Add current BrowserStack projects to sln

Commit:
Bump BrowserStack tools version

Commit:
Merge branch 'main' into use-browserstack-for-appium-tests

Commit:
Remove browserstack.yml since now moved

Commit:
Bump BrowserStack dependencies

Commit:
Update paths

jfversluis commented:
This seems stale by now and we’re not enabling this anytime soon. Thanks for the effort!
